_posts/en/posts/2024-09-18-disclose-headers-oom.md

```diff
@@ -0,0 +1,112 @@
+2. It relies on the ecosystem semi-regularly adopting updated software with new checkpoints, a
+   practice which Bitcoin Core contributors have long been uncomfortable with.
+
+It later got increased attention when Braydon Fuller [posted his "Chain width expansion"
+writeup](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2019-October/017354.html) to the
+bitcoin-dev mailing list in October 2019. He had previously responsibly reported it to the Bitcoin
+Core security list. The suggested approach was not adopted in Bitcoin Core due to concerns about
+network convergence when limiting the number of parallel chains.
+
+At the time, the computational cost of creating a huge low-difficulty headers chain was equal to
+about 32.28% of mining one block at the tip. That is a cost of about 4.12 BTC since the block reward
+then was about 12.77 BTC.
+
+By February 2022, the cost of the attack had dropped further to around 14.73% of the cost of mining
+a block, and this prompted investigation of alternative solutions. If unaddressed, the cost today
+(September 2024) would just be 4.44% of a block. These figures translate to a cost of about 1.07 BTC
+and 0.14 BTC respectively, given the block reward at these dates.
+
+A protection against this DoS was implemented in Bitcoin Core PR
+[#25717](https://github.com/bitcoin/bitcoin/pull/25717), whereby the node will first verify a
+presented chain has enough work before committing to store it. With that, Bitcoin Core no longer
+relies on having checkpoints to protect against any known attacks.
+
+## Attribution
+
+Credit goes to David Jaenson and Braydon Fuller for independently re-discovering the attack,
+estimating its cost and suggesting modifications.
+
+Credit goes to Suhas Daftuar and Pieter Wuille for researching a satisfying fix and implementing it.
+
+## Timeline
+
+* 2010-07-17 - Bitcoin 0.3.2 is released, which introduces checkpoints. They protect among other
+  things against low-difficulty block spam.
+* 2011-11-21 - Bitcoin 0.5.0 is released, which skips script validation for blocks before the last
+  checkpoint. This makes the role of checkpoints even more security-critical.
+* 2014-04-09 - Block 295000 is mined, which becomes the last Bitcoin Core checkpoint. The protection
+  offered by checkpoints against block spam starts eroding from this point on as hashrate costs
+  decrease.
+* 2015-02-16 - Bitcoin Core 0.10.0 is released, with headers-first synchronization. This weakens the
+  low-difficulty block spam attack to a block *header* spam attack.
+* 2017-03-08 - Bitcoin Core 0.14.0 is released, which disentangles the skipping of script validation
+  from checkpoints, leaving them only relevant for protecting against block header spam.
+* 2019-01-28 - David Jaenson reports this issue to the Bitcoin Core security mailing list.
+* 2019-09-18 - Braydon Fuller emails the Bitcoin Core security list with a paper titled "Bitcoin
```

Comment:
https://bcoin.io/papers/bitcoin-chain-expansion.pdf

If you feel like linking

Reply:
Sure, had also linked the email already above.